File: app/backend/models/crashed_container.py
from sqlalchemy import Column, Integer, String, DateTime
from app.backend.core.database import Base


class CrashedContainer(Base):
    __tablename__ = 'crashedcontainers'
    
    id = Column(Integer, primary_key=True)
    container_id = Column(String(100))
    container_name = Column(String(100))
    logs = Column(String(5000), nullable=True)
    crashedon = Column(DateTime, nullable=False)
 
    def __repr__(self):
        return f"<CrashedContainer(name= '{self.container_name}', containerId='{self.container_id}')>"

File: app/dashboard/src/models/crashedContainer.ts
export interface CrashedContainerBase {
    container_id: string;
    container_name: string;
    crashed_on?: string | null;
}

export interface CrashedContainerLogs extends CrashedContainerBase {
    logs: string;

}

export interface CrashedContainerChartStats extends CrashedContainerBase {
    crash_count: number;
}